### PR TITLE
Tree-sitter-darklang: Fix let_expression to handle lp_tuple correctly

### PR DESCRIPTION
**Pull request by OceanOak**
No changelog

### CODE REVIEW

File: tree-sitter-darklang/grammar.js [hunk 1]
```diff
@@ -924,6 +924,7 @@ module.exports = grammar({
     //
     // Tuple
     tuple_type_reference: $ =>
+      // CLEANUP: make the parens optional
```
Comment (StachuDotNet) [inline tree-sitter-darklang/grammar.js:927]: We can leave this CLEANUP in but I don't hate the idea of requiring `Json.parse<(Int * Bool)>`, not allowing `Json.parse<Int * Bool>` - the former feels clearer. What do you think?

Comment (StachuDotNet) [reply tree-sitter-darklang/grammar.js:927]: actually, "tolerable parser" so nvm - the CLEANUP is good and we should allow the latter too. But maybe I'd prefer to _prettify_ the TypeRef with the parens :)

File: packages/darklang/languageTools/parser/expr.dark [hunk 2]
```diff
@@ -399,15 +399,50 @@ module Darklang =
...
-            (WrittenTypes.Expr.ELet(
-              node.range,
-              WrittenTypes.LetPattern.LPVariable(identifier.range, identifier.text),
-              expr,
-              body,
-              keywordLet.range,
-              symbolEquals.range
-            ))
-            |> Stdlib.Result.Result.Ok
+            match identifier.children with
```
Comment (StachuDotNet) [inline packages/darklang/languageTools/parser/expr.dark:402]: Funny we've had the `let_pattern` in the grammar for a while, not used by `let`s!
Given the fact that let patterns are used by both `ELet` and `ELambda`, this should probably be extracted into a `LetPattern.parse` (or another name - `parseLetPattern` is probably OK based on your judgment)

Comment (OceanOak) [reply packages/darklang/languageTools/parser/expr.dark]: we had it, I just forgot about it :woman_facepalming: 
Fixed

**Pull request merged**